tools/bbmap.py

```diff
@@ -28,38 +31,223 @@ def __init__(self, install_methods=None):
+                #   and clump them all together as unpaired.
+                if inFastq2 is None or os.path.getsize(inFastq2) < 10:
+                    self.execute(tool='clumpify.sh', 
+                                    in1=inFastq1,
+                                    out=outFastq1,
+                                    dedupe=True,
+                                    subs=subs,
+                                    passes=passes,
+                                    dupedist=dupedist,
+                                    k=kmer_size,
+                                    optical=optical,
+                                    spany=spany,
+                                    spanx=spanx,
+                                    adjacent=adjacent,
+                                    usetmpdir=True,
+                                    tmpdir=t_dir,
+                                    # if reads should be treated as unpaired, both 'unpair','repair' should be set to True
+                                    unpair=treat_as_unpaired,
+                                    repair=treat_as_unpaired,
+                                    containment=containment,
+                                    JVMmemory=JVMmemory,
+                                    **kwargs)
+                else:
+                    self.execute(tool='clumpify.sh', 
+                                    in1=inFastq1, in2=inFastq2,
+                                    out=outFastq1, out2=outFastq2,
+                                    dedupe=True,
+                                    subs=subs,
+                                    passes=passes,
+                                    dupedist=dupedist,
+                                    k=kmer_size,
+                                    optical=optical,
+                                    spany=spany,
+                                    spanx=spanx,
+                                    adjacent=adjacent,
+                                    usetmpdir=True,
+                                    tmpdir=t_dir,
+                                    # if reads should be treated as unpaired, both 'unpair','repair' should be set to True
+                                    unpair=treat_as_unpaired,
+                                    repair=treat_as_unpaired,
+                                    containment=containment,
+                                    JVMmemory=JVMmemory,
+                                    **kwargs)
+                # Make a list of reads to keep
+                with open(per_lb_read_list, 'at') as outf:
+                    for fq in (outFastq1, outFastq2):
+                        with util.file.open_or_gzopen(fq, 'rt') as inf:
+                            line_num = 0
+                            for line in inf:
+                                if (line_num % 4) == 0:
+                                    idVal = line.rstrip('\n')[1:]
+                                    if idVal.endswith('/1'):
```

Comment:
Do you have an example of what you mean? NovaSeq-like read IDs look the same from a few bam files I compared (NovaSeq vs HiSeq). Since we're doing the conversion to fastq files it should hopefully be fine? This is also the same code we've been using for mvicuna dedup, so the behavior should be the same (though that's not to say we missed a code path to update for NovaSeq support).

Reply:
Read IDs that identify the pairs with a space a `1:N:0:barcodes` instead of `/1` or `/2`
![image](https://user-images.githubusercontent.com/601839/70067792-cfa5b580-15bc-11ea-963d-1ba8aec251b2.png)